### PR TITLE
Add support for twig embedding

### PR DESCRIPTION
**Pull request by cs-shadow**
d.o issue: https://www.drupal.org/node/2279257

Done some head chasing. Still need to do following things:
- [x] Support UUID
- [x] Support display plugins
- [x] Add tests for display plugins and settings

### CODE REVIEW

**Comment by cs-shadow** [comment]
I'm dropping support for view_mode at the moment in favor of display plugins.

**Comment by cs-shadow** [comment]
@davereid Would like to get your feedback on this one?

**Comment by davereid** [comment]
Yes, I'd love to take a deep look at this tomorrow.

File: src/Twig/EntityEmbedTwigExtension.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+   *   The entity ID or entity UUID.
+   * @param string $display_plugin
+   *   (optional) The display plugin to be used to render the entity.
+   * @param array $display_settings
+   *   (optional) A list of settings for the display plugin.
+   *
+   * @return array
+   *   A render array from entity_view().
+   */
+  public function getRenderArray($entity_type, $entity_id, $display_plugin = 'default', $display_settings = array()) {
```
Comment (davereid) [inline src/Twig/EntityEmbedTwigExtension.php]: We should typehint `array $display_settings = []`?

**Comment by davereid** [comment]
@cs-shadow Is this PR missing the actual test file?

**Comment by cs-shadow** [comment]
@davereid Yep, the test file was missing, my bad. Added it.

File: tests/modules/entity_embed_twig_test/entity_embed_twig_test.info.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: 'Entity Embed twig test'
```
Comment (davereid) [inline tests/modules/entity_embed_twig_test/entity_embed_twig_test.info.yml]: A little curious why we don't just merge this into entity_embed_test module itself?

**Comment by davereid** [comment]
I rebased this together on https://github.com/drupal-media/entity_embed/pull/172 to get a little cleaner history. Going to merge that issue.

**Pull request closed without merging**